src/components/Identity/createComponent.js

```diff
@@ -78,50 +74,46 @@ export default (idSyncs, config, logger, cookieJar, optIn, eventManager) => {
 
           let promise;
 
-          if (ecid) {
-            addIdsContext(payload, ecid);
-          } else if (deferredForEcid) {
-            // We don't have an ECID, but the first request has gone out to
-            // fetch it. We must wait for the response to come back with the
-            // ECID before we can apply it to this payload.
-            logger.log("Delaying request while retrieving ECID from server.");
-            promise = deferredForEcid.promise.then(() => {
-              logger.log("Resuming previously delayed request.");
-              addIdsContext(payload, getEcid());
-            });
-          } else {
-            // We don't have an ECID and no request has gone out to fetch it.
-            // We won't apply the ECID to this request, but we'll set up a
-            // promise so that future requests can know when the ECID has returned.
-            deferredForEcid = defer();
-            payload.expectResponse();
-            if (
-              config.thirdPartyCookiesEnabled &&
-              areThirdPartyCookiesSupported(getBrowser(window))
-            ) {
-              payload.useIdThirdPartyDomain();
+          if (!ecid) {
+            if (deferredForEcid) {
+              // We don't have an ECID, but the first request has gone out to
+              // fetch it. We must wait for the response to come back with the
+              // ECID before we can apply it to this payload.
+              logger.log("Delaying request while retrieving ECID from server.");
+              promise = deferredForEcid.promise.then(() => {
+                logger.log("Resuming previously delayed request.");
+              });
+            } else {
+              // We don't have an ECID and no request has gone out to fetch it.
+              // We won't apply the ECID to this request, but we'll set up a
+              // promise so that future requests can know when the ECID has returned.
+              deferredForEcid = defer();
+              payload.expectResponse();
+              if (
+                config.thirdPartyCookiesEnabled &&
```

Comment:
Should we add an inline comment explaining what `payload.useIdThirdPartyDomain` is?

Reply:
Yes, I'll do that.

Reply:
Done. I think this is what you were describing.

Reply:
Perfect!